platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/DefaultVisitedArtifactSet.java

```diff
@@ -45,8 +45,8 @@ public DefaultVisitedArtifactSet(
     }
 
     @Override
-    public SelectedArtifactSet select(ArtifactSelectionSpec spec) {
-        SelectedArtifactResults artifacts = artifactsResults.select(artifactVariantSelector, spec, false);
+    public SelectedArtifactSet select(ArtifactSelectionSpec spec, boolean lenient) {
+        SelectedArtifactResults artifacts = artifactsResults.select(artifactVariantSelector, spec, lenient);
```

Comment:
❌ I think we can revert this and (hopefully) all of the `lenient` wiring you've done here -- which should resolve your two comments about if we should pass lenient or not in some cases. 

The reasoning being is that this is the legacy way of specifying leniency. `LenientConfiguration` uses this to be lenient -- though with modern constructs (`ResolutionBackedFileCollection`) we specify leniency by either ignoring or rethrowing exceptions at the time of visiting these selected results. 

Stated more concisely, legacy mechanism request leniently during selection. Modern mechanism ask selection to be strict, then ignore the failures later if its a lenient resolution.